spikingjelly/activation_based/triton_kernel/flex_sn_inductor/kernel.py

```
@@ -0,0 +1,174 @@
+def _diff_mask(
+    core_fn: Callable,
+    num_outputs: int,
+    num_states: int,
+    example_inputs: Tuple[torch.Tensor, ...],
+) -> List[bool]:
+    """Return which of core_fn's outputs are differentiable."""
+    ex = [t.clone().detach().requires_grad_(True) for t in example_inputs]
+    try:
+        with torch.enable_grad():
+            outs = core_fn(*ex)
+    except Exception:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)